### DISCUSSION TITLE
Finalynx will need to show graphs soon, where to show them?

### DISCUSSION THREAD

**Discussion by MadeInPierre** [MadeInPierre/finalynx#28] [General]
Hi! 

Finalynx will soon show your asset allocation on a per-folder level and much more: #27

However, I'm not sure how to output them...
- Should Finalynx remain a pure command-line tool and plot graphs with tools such as uniplot?
- Should Finalynx output an external report (such as a PDF)? Jupyter Notebook?
- Should Finalynx become a [web dashboard](https://themes.getbootstrap.com/preview/?theme_id=19799)? (I have no experience in web dev)

I think some plots could stay in the terminal (e.g. global asset allocation), but it may become cluttered with many graphs and more difficult to interact with. A web dashboard is nice but more difficult to develop (especially for me), and may interfere with Finary's official webapp.

Any opinions appreciated!

**Comment by nmathey** [comment]
Hi

I vote for offline generated document (PDF or HTML5) .
This will be more portable than having a python environement to run the code.

IMO Terminal should stay focus on managing the portfolio content/allocation.

**Reply by MadeInPierre** [reply]
En effet, je trouve ça un peu étrange d'afficher les résultats sur plusieurs endroits, mais j'ai l'impression que ça sera nécessaire de toute façon :sweat_smile: 

Vu que je mettrais probablement trop de temps à créer un dashboard web, je propose de rendre Finalynx extensible pour pouvoir ajouter des IHM dessus, et je commencerai par un output PDF qui est plus simple pour moi.

Je verrais bien la répartition d'informations suivantes : 
- **Command-line :**
    - Portfolio complet avec hiérarchie
    - Recommendations à côté de chaque ligne venant de `copilot` à propos de combien investir pour atteindre les objectifs
    - Statistiques générales en dessous de l'arbre (e.g. allocation globale par classe d'actifs, résumé textuel de chaque graphique)
- **Output PDF :**
    - On indique dans la config les graphiques qui nous intéressent, ils sont générés dans le PDF 
    - Graphique principal de simulation de vie avec événements
    - Rapport de recommendations
- **Output Web :**
    - Rendre le tout interactif, on clique sur une ligne de l'arbre et tous les graphiques se mettent à jour sur ce nouveau scope, on crée des événements, on compare le résultat sur plusieurs événements pour mieux planifier, ...

Si jamais le web te tente... :wink:

**Reply by MadeInPierre** [reply]
Je viens de découvrir [ce package](https://nicegui.io/) et ça a l'air parfait pour Finalynx! :smile: (créer un site web avec uniquement du code Python, ça correspond à mes compétences perso)

**Reply by nmathey** [reply]
En phase avec toi! Et oui NiceGUI semble être un bon compromis pour moi également. Je vais essayer de jouer avec de mon côté :)
